ci/jenkins/jobs/macros.yaml

```diff
@@ -259,6 +259,7 @@
           VERSION="$JOB_NAME-$BUILD_NUMBER" make
 
           sed -i "s|#serviceCIDR: 10.96.0.0/12|serviceCIDR: 100.64.0.0/13|g" build/yamls/antrea.yml
+          sed -i 's|#enablePrometheusMetrics: false|enablePrometheusMetrics: true|g' build/yamls/antrea.yml
```

Comment:
I had this issue too when writing e2e tests for selfSignedCert=false case which is also not the default configuration and wondered whether I should change the deployment yaml or update the config when running the test case.
Finally I chose the former as changing deployment yaml is not flexible enough as it must be hardcoded true or false. I would suggest testing it by updating configmap at runtime like https://github.com/vmware-tanzu/antrea/blob/master/test/e2e/security_test.go#L45-L67, so we can avoid introducing extra argument and document to run the prometheus tests, otherwise `push_antrea.sh` and `go test` may have to take many arguments after we introduce more features.

Reply:
Honestly I don't see a reason not to enable Prometheus by default. However the flag will still be required as some of the tests require deployment of Prometheus server.
As this has resources impact, we would need a flag anyway (or can we do that programmatically as well?)

Reply:
I didn't know we tested the Prometheus server, thought antrea's scope is guaranteeing the desired metrics are present (and correct if possible) in the metrics response. 
My intention was reducing the preparation work of running tests and verifying the essential effect of antrea's code so that the code can be ran easily and quickly, with less environment dependency. 
For example, when we run upstream tests, we just need to care about some generic parameter "focus", "skip", "parallel" to choose what to run and how to run, not having to care about how to setup the environment to run specific tests and have a separate argument for them. My concern is after adding the first "-prometheus" argument, will this pattern be followed leading to more "-XXX" argument to run the whole tests in future?
Let's also see what @antoninbas @jianjuns would say.

Reply:
@tnqn your points make sense and we need to work on this. I think we can go with the current solution for now (v0.7.0) and take more time to come up about a better solution. This specific case is a bit different because we actually want to test the sample Prometheus manifest (antrea-prometheus.yml), and not just Antrea's ability to publish these metrics. That may warrant the use of this additional flag until we can think of something better. If it were just about testing the Antrea `/metrics` endpoint, I agree that we could either unconditionally enable them or update the configMap in the test (even though at the moment it would require restarting all agents & the controller).

I suggest we merge this now, and then look into the following:
* for Antrea `/metrics` tests, make sure metrics are enabled (through configMap) and run tests unconditionally
* for tests that require the Prometheus server, maybe the yaml can be applied programmatically by the tests (and deleted at the end). We can group all the server tests (with shared setup & clenaup) into one and use subtests. It still makes sense IMO to *not* run these tests by default, so maybe we have to stick with the flag.

We can also look into whether we can categorize tests like upstream does. Ideally without having to switch to ginkgo.

Reply:
The plan makes sense to me. Thanks.